### PR TITLE
Dadguide rewrite Graph Addition

### PR DESCRIPTION
**Pull request by chasehult**

### CODE REVIEW

File: dadguide/database_manager.py [hunk 1]
```diff
@@ -201,10 +219,79 @@ def _get_table_fields(self, table_name: str):
...
+        to_check = {monster_id}
+        while to_check:
+            mid = to_check.pop()
+            if mid in ids:
+                continue
+            n = self.graph[mid]
+            to_check.update(get_edges(n, 'evolution'))
+            to_check.update(get_edges(n, 'transformation'))
+            to_check.update(get_edges(n, 'back_evolution'))
+            to_check.update(get_edges(n, 'back_transformation'))
```
Comment (TsubakiBotPad) [inline dadguide/database_manager.py:287]: doesn't this screw up dark magician girl and winged dragon? e.g `^id equip DMG`

File: dadguide/database_manager.py [hunk 2]
```diff
@@ -201,10 +219,79 @@ def _get_table_fields(self, table_name: str):
...
+        es = self._query_many("SELECT * FROM evolutions", (), DictWithAttrAccess)
+        aws = self._query_many("SELECT * FROM awakenings", (), DgAwakening)
+        lss = self._query_many("SELECT * FROM leader_skills", (), DgLeaderSkill, idx_key='leader_skill_id')
+        ass = self._query_many("SELECT * FROM active_skills", (), DgActiveSkill, idx_key='active_skill_id')
+        ss = self._query_many("SELECT * FROM series", (), DgSeries, idx_key='series_id')
+
+        for m in ms:
+            self.graph.add_node(m.monster_id, **m,
+                                    dgawakenings=[],
+                                    dgls=lss.get(m.leader_skill_id),
```
Comment (TsubakiBotPad) [inline dadguide/database_manager.py:235]: would recommend not abbreviating, and also remove dg prefix cuz dadguide literally means jack shit

File: padvision/__init__.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from .padvision import PadVision
+from .padvision import PadVision, NeuralClassifierBoardExtractor
```
Comment (TsubakiBotPad) [inline padvision/__init__.py:1]: unrelated?

File: dadguide/__init__.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
-from .dadguide import Dadguide
+from .dadguide import *
```
Comment (TsubakiBotPad) [inline dadguide/__init__.py:1]: ?

File: dadguide/dadguide.py [hunk 5]
```diff
@@ -67,7 +67,7 @@ def __init__(self, bot, *args, **kwargs):
         # Map of google-translated JP names to EN names
         self.translated_names = {}
 
-        self.database = load_database(None)
+        self.database = None # load_database(None)
```
Comment (TsubakiBotPad) [inline dadguide/dadguide.py:70]: did you mean to comment this out and rely on the reload?

File: dadguide/database_manager.py [hunk 6]
```diff
@@ -436,15 +454,25 @@ def get_all_monster_ids_query(self, as_generator=True):
 
     def refresh_monsters(self):
         if self.expiry < datetime.now().timestamp():
-            self.expiry = int(datetime.now().timestamp()) + 60*60
-            self.cachedmonsters = {}
+            #self.expiry = int(datetime.now().timestamp()) + 60*60
```
Comment (TsubakiBotPad) [inline dadguide/database_manager.py:457]: remove?

File: dadguide/database_manager.py [hunk 7]
```diff
@@ -96,15 +99,23 @@ def __init__(self, data_file):
         self._con = lite.connect(data_file, detect_types=lite.PARSE_DECLTYPES)
         self._con.row_factory = lite.Row
 
+        self.graph = None
+        self.build_graph()
 
         self.cachedmonsters = None
         self.expiry = 0
+        self.generate_all_monsters()
```
Comment (TsubakiBotPad) [inline dadguide/database_manager.py:107]: in a subsequent pr, we should try to rip apart DadguideDatabase so that its primary function is only to query sqlite. Ideally we would change all the call sites of `get_active_skill_query`, `get_leader_skill_query`... etc to call the graph (which would have these functions) directly.

File: dadguide/__init__.py (hunk 4 above)
Comment (chasehult) [reply dadguide/__init__.py:1]: Necessary for PDE and also for testing things out with ^eval

File: dadguide/database_manager.py (hunk 1 above)
Comment (chasehult) [reply dadguide/database_manager.py:287]: Yes, but I'm going to cross that bridge when it it comes up.  This seems like the best right now, because we don't know what direction GH is gonna take with transforms that can be obtained.  I can change it to do what would be best for specifically Winged Dragon's tree though.

File: padvision/__init__.py (hunk 3 above)
Comment (chasehult) [reply padvision/__init__.py:1]: Same issue as the `from .dadguide import *`, but less related to the whole thing.
